cfme/services/catalogs/orchestration_template.py

```diff
@@ -1,166 +1,284 @@
+
+    add_button = Button("Save")
+
+    @property
+    def is_displayed(self):
+        return (
+            self.title.text == 'Adding a new Service Dialog from '
+                               'Orchestration Template "{}"'.format(self.obj.name) and
+            self.orchestration_templates.is_opened
+        )
+
+
+class OrchestrationTemplate(Updateable, Pretty, Navigatable):
+
+    def __init__(self, template_type=None, template_name=None, description=None,
+                 appliance=None):
+        Navigatable.__init__(self, appliance)
         self.template_type = template_type
         self.template_name = template_name
         self.description = description
 
-    def create_form(self, content):
-        sel.force_navigate('create_new_template',
-                           context={'template_type': self.template_type})
+    def create(self, content):
+        view = navigate_to(self, "AddTemplate")
         if(self.template_type == "CloudFormation Templates"):
             temp_type = "Amazon CloudFormation"
         else:
             temp_type = "OpenStack Heat"
-        fill(create_template_form, {'description': self.description,
-                                    'template_type': temp_type,
-                                    'content': content,
-                                    'template_name': self.template_name},
-            action=create_template_form.add_button)
-
-    def create(self, content):
-        self.create_form(content)
-        try:
-            flash.assert_success_message('Orchestration Template "{}" was saved'.format(
-                self.template_name))
-        except Exception as e:
-            if error.match("Error during 'Orchestration Template creation': Validation failed: \
-                            Md5 of content already exists (content must be unique)", e):
-                self.create_form(content + " ")
-            raise
+        view.fill({'name': self.template_name,
+                   'description': self.description,
+                   'template_type': temp_type,
+                   'content': content})
+        view.add_button.click()
+        view.flash.assert_no_error()
+        view.flash.assert_message('Orchestration Template '
+                                  '"{}" was saved'.format(self.template_name))
 
     def update(self, updates):
-        sel.force_navigate('edit_template',
-                           context={'template_type': self.template_type,
-                                    'template_name': self.template_name})
-        fill(create_template_form, {'template_name': updates.get('template_name', None),
-                                    'description': updates.get('description', None)},
-             action=create_template_form.edit_button)
-        flash.assert_success_message('Orchestration Template "{}" was saved'.format(
+        view = navigate_to(self, "Edit")
+        view.fill({'description': updates.get('description', None),
```

Comment:
@psav  i'm wondering if views should get fitting update methods that handle their saving and message checks 
 
it would then read as `edit_view.update(description=..., name=...)`

that method would do the fill, save and assert_message dance

this would encapsulate the action behavior of the view with the view, allowing code using it to be less concerned about their structure